### PR TITLE
dealing with RF/RFutils (#95)

### PR DESCRIPTION
**Pull request by achubaty**
- provide these pkgs via r-universe repo (should be changed to use ropensci's r-universe)
- mv these pkgs to Suggests
- use these pkg conditionally in functions and tests

- fixed other R CMD check problems along the way

### CODE REVIEW

File: R/NLMR.R [hunk 1]
```diff
@@ -19,5 +19,13 @@
 
 globalVariables(c("x", "value", ".", "ctf"))
 
+.cranLikeRepo <- "https://predictiveecology.r-universe.dev/"
+.messageRandomFields <- paste(
+  "Package 'RandomFields' is required, but no longer available on CRAN.",
+  "Try installing the most recent archived version:",
+  sprintf("  install.packages('RandomFields', repos = '%s')", .cranLikeRepo),
```
Comment (bitbacchus) [inline R/NLMR.R:26]: should we suggest installing RandomFieldsUtils here, too?

**Review by bitbacchus — APPROVED** [review]
Looks great to me! I just wonder whether we should suggest installing RandomFieldsUtils, too? Anything else from your side @marcosci?

**Comment by achubaty** [comment]
Since RF depends on RFutils, it's not strictly necessary to depends on RFutils.

File: R/nlm_mosaicfield.R [hunk 2]
```diff
@@ -57,142 +56,147 @@ nlm_mosaicfield <- function(ncol,
...
-    )
-
-    if (isTRUE(collect)) {
-      mosaicfield_list <- list()
-      mosaicfield_list[[1]] <- mosaicfield_result
-    }
 
-    for (i in 2:n) {
-      mosaicfield_n <- spatstat.random::rMosaicField(
+  if (requireNamespace("RandomFieldsUtils", quietly = TRUE)) {
```
Comment (bitbacchus) [inline R/nlm_mosaicfield.R:60]: We should check here for `RandomFields`, not `RandomFieldsUtils`, right?

Comment (achubaty) [reply R/nlm_mosaicfield.R:60]: No, because it's only using RandomFieldsUtils not RandomFields.

Comment (bitbacchus) [reply R/nlm_mosaicfield.R:60]: Ahhh, I see. Thanks :-)

**Comment by achubaty** [comment]
@bitbacchus I just pushed an update to this PR that explicitly adds the additional repository to the 2 failing GitHub Actions configs, which should hopefully resolve those failures. (Though note I switched to v2 of the r-setup from v1, without updating the downstream Actions -- i.e., there may be more tweaking of the Actions config needed)

**Comment by achubaty** [comment]
updated all R actions to v2 -- not sure why it wasn't finding the packages in the r-universe repo, because they are there and working on other test systems :/

**Comment by mhesselbarth** [comment]
Seems like it's still not finding the packages unfortunately

**Comment by achubaty** [comment]
:cry: looks like a bug in `pak::pkg_install()` which is being used in that action.

**Comment by achubaty** [comment]
@mhesselbarth finally figured it out. `pak` choked on the trailing slash in the repo url (https://github.com/r-lib/pak/issues/400).

Removing the slash is worknig for me locally now, and I've pushed 7702bf5963b7ca4fdf3191eee9323f4e7ffccf96 to deal with this.

**Comment by bitbacchus** [comment]
Hm, the build fails for Linux and macOS. Apparently, the XML package can't be installed on Linux, but I don't see an obvious reason at the moment. At least, `libxml2-dev` seems to install successfully.

**Comment by achubaty** [comment]
perhaps a weird blip? my other GitHub Actions workflows that install `XML` on Linux are working.

**Comment by bitbacchus** [comment]
The issue seems to be with `pak` and R 4.2: https://github.com/r-lib/actions/issues/559 

Can you try if this helps?
```r
extra-packages: any::rcmdcheck, any::XML
```

Thanks a lot for your effort!

**Comment by achubaty** [comment]
@bitbacchus thanks for tracking down the issue in `pak`. I've pushed the workaround for `XML`. :crossed_fingers:

**Comment by bitbacchus** [comment]
macOS still fails:

Screenshot from 2022-08-11 19-23-08

Why though?

**Comment by mhesselbarth** [comment]
I tried to install RandomFields locally on my macOS, and needed to manually install gfortran because apparently thats shipped differently with the newest gcc compiler updates which caused issue. Maybe somehow related to that?

(https://github.com/fxcoudert/gfortran-for-macOS)

**Comment by achubaty** [comment]
The `brew` setup should automatically be getting the compiler tools needed, so I don't think that's the issue -- on my macOS machine, `pak` is trying to get the pkg from CRAN (and failing) but isn't trying the r-universe repo for some reason.

I'm going to try manual install.

**Comment by achubaty** [comment]
@mhesselbarth @bitbacchus this is now passing for me :tada:

**Comment by bitbacchus** [comment]
Awesome, tanks a ton, @achubaty :-)

**Comment by achubaty** [comment]
Thanks - any idea how soon you plan to get updated package version to CRAN?

**Pull request merged**